add Sec-CH-Prefers-Reduced-Motion content (#22243)
mdn/content: files/en-us/web/http/client_hints/index.md

The decision might be based on user preferences, a permission request, or the permission policy.
All client hints that are not low entropy hints are considered high entropy hints.

> ## Specifying critical client hints

Reviewer: ```suggestion
## Critical client hints
```

The decision might be based on user preferences, a permission request, or the permission policy.
All client hints that are not low entropy hints are considered high entropy hints.

> ## Critical client hints
> 

Reviewer: ```suggestion
## Critical client hints

A _critical client hint_ is one where applying the response may significantly change the rendered page, potentially in a way that is jarring or will significantly affect usability, and which must therefore be applied before the content is rendered.
For example, `Sec-CH-Prefers-Reduced-Motion` is commonly treated as a critical hint, because it might markedly affect the behaviour of animations, and because a user who chooses this preference really needs for it to be set.

```

> You can use the {{HTTPHeader("Critical-CH")}} response header along with `Accept-CH` to specify that an accepted client hint is also a _critical client hint_. User agents should automatically send the server that client hint to indicate their preference in that regard.

Reviewer: FWIW "
- You" better as "server" or "user agent" when documenting HTTP. Otherwise the flow can be confusing.
- "User agents should automatically send the server that client hint to indicate their preference in that regard." is confusing/too early. 

```suggestion
A server can use the {{HTTPHeader("Critical-CH")}} response header along with `Accept-CH` to specify that an accepted client hint is also a critical client hint (note that a header in `Critical-CH` must also appear in `Accept-CH`).
User agents receiving a response with `Critical-CH` must check if the indicated critical headers were sent in the original request. If not, then the user agent will retry the request rather than rendering the page.
This approach ensures that client preferences set using critical client hints are always used, even if not included in the first request, or if the server configuration changes.

```

> For example, in this case, the server tells a client via {{httpheader("Accept-CH")}} that it accepts `Sec-CH-Prefers-Reduced-Motion`. {{httpheader("Critical-CH")}} is also used to specify that `Sec-CH-Prefers-Reduced-Motion` is considered a critical client hint:

Reviewer: ```suggestion
For example, in this case, the server tells a client via {{httpheader("Accept-CH")}} that it accepts `Sec-CH-Prefers-Reduced-Motion`, and {{httpheader("Critical-CH")}} is used to specify that `Sec-CH-Prefers-Reduced-Motion` is considered a critical client hint:
```

```http
HTTP/1.1 200 OK
Content-Type: text/html
Accept-CH: Sec-CH-Prefers-Reduced-Motion
Critical-CH: Sec-CH-Prefers-Reduced-Motion
```

> The client automatically retries the request, telling the server via `Sec-CH-Prefers-Reduced-Motion` that it has a user preference for reduced-motion animations.

Reviewer: ```suggestion
As `Sec-CH-Prefers-Reduced-Motion` is a critical hint that was not in the original request, the client automatically retries the request — this time telling the server via `Sec-CH-Prefers-Reduced-Motion` that it has a user preference for reduced-motion animations.
```

The decision might be based on user preferences, a permission request, or the permission policy.
All client hints that are not low entropy hints are considered high entropy hints.

> ## Critical client hints

Reviewer: FYI, I have rewritten this to explain how client hints work. We could go further with all of this page to show sequence diagrams etc, but IMO this is all pretty clear.

Can you recheck it?

A _critical client hint_ is one where applying the response may significantly change the rendered page, potentially in a way that is jarring or will significantly affect usability, and which must therefore be applied before the content is rendered.

> For example, `Sec-CH-Prefers-Reduced-Motion` is commonly treated as a critical hint, because it might markedly affect the behaviour of animations, and because a user who chooses this preference really needs for it to be set.

Reviewer 2: > `Sec-CH-Prefers-Reduced-Motion` is commonly treated as a critical hint

Interesting, is this observed in the wild? I had seen only a few sites using `Sec-CH-Prefers-Color-Scheme` to avoid ["flash of unstyled content"](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) typically flash of white content in dark color scheme) but I have never seen any issues related to animations.

Reviewer: No, this is creative writing. But it should be a critical hint - if a user sets a preference for reduced motion it is usually because they have an actual health risk. So this is my way of saying that it is optional, but a server that provides this kind of information should set it as critical.

> UA client hints provide a more efficient and privacy preserving way of getting the desired information.
> They are eventually expected to replace this older approach.

> ### User preference media features client hints

Reviewer: FYI, moved this into its own section below. This is a useful way of presenting the sets as they come along.

Reviewer: > To begin with, I've added a line about this specific classic of UA hints to the "User-agent client hints" section of the Client hints page, as they are definitely in that category. That'll be in the next commit.

PS Reasoning is that yes, they are user agent hints, but they are quite specific in the way they are set and how someone might look for them (i.e. when you compare them to the other hints in that category). So this is a useful grouping in the same ways as device hints are.

Author: Yup, this looks good.
